test/writer/model_reader_test.dart

```diff
@@ -66,6 +66,76 @@ TestModel:
       expect(simpleDateTime.type, isA<DateTimeType>());
       expect(simpleDateTime.isRequired, false);
     });
+
+    test('Test required not definable anymore', () {
+      dynamic error;
+      try {
+        YmlGeneratorConfig(
+                PubspecConfig("name: test"),
+                """
+TestModel:
+  properties:
+    requiredString:
+        type: String
+        required: true
+    optionalString:
+        type: String
+        required: false
+""",
+                '')
+            .models;
+      } catch (e) {
+        error = e;
+      }
+
+      expect(error, isNotNull);
```

Comment:
check if the error is in fact related to the required:true so check the actual error not just `isNotNull`